### ISSUE TITLE
Making a class static is suggested for base classes

### ISSUE THREAD

**Issue by ddobrev** [JosefPihrt/Roslynator#363]
If a class only has static members but is also sub-classed, there should be no suggestion to make it static.

**Comment by JosefPihrt** [comment]
I assume that you mean this:
```csharp
public class B
{
    public static void M()
    {
    }
}

public class C : B
{
}
```
RCS1102 can reports a false positive in certain rare cases such as yours. Another example is https://github.com/JosefPihrt/Roslynator/issues/358. This behavior is by design and is based on the fact that analyzer does not (and cannot) check all references.

By the way, CA1052 is also reported.

My recommendation is to suppress the diagnostic:
``` csharp
[SuppressMessage("Design", "RCS1102", Justification = "<Pending>")]
public class B
```
